### PR TITLE
Create vernam_cipher.py

### PR DESCRIPTION
**Pull request by Anupamaraie**
added vernam cipher

### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: ciphers/vernam_cipher.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import doctest
+
+
```
Comment (cclauss) [inline ciphers/vernam_cipher.py:3]:
```suggestion
```

File: ciphers/vernam_cipher.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+encrypted_text = vernam_encrypt(plaintext, key)
+decrypted_text = vernam_decrypt(encrypted_text, key)
+print("\n\n")
+print("Plaintext:", plaintext)
+print("Encrypted:", encrypted_text)
+print("Decrypted:", decrypted_text)
+
+
+if __name__ == "__main__":
+    doctest.testmod()
```
Comment (cclauss) [inline ciphers/vernam_cipher.py:44]:
```suggestion
if __name__ == "__main__":
    from doctest import testmod
    doctest.testmod()

    # Example usage
    plaintext = input("Enter the message: ").upper()
    key = input("Enter the key: ").upper()
    encrypted_text = vernam_encrypt(plaintext, key)
    decrypted_text = vernam_decrypt(encrypted_text, key)
    print("\n\n")
    print("Plaintext:", plaintext)
    print("Encrypted:", encrypted_text)
    print("Decrypted:", decrypted_text)
```

**Review by cclauss — CHANGES_REQUESTED** [review]

File: ciphers/vernam_cipher.py (hunk 2 above)
Comment (Anupamaraie) [reply ciphers/vernam_cipher.py]: ruff is failing when I do this

File: ciphers/vernam_cipher.py (hunk 1 above)
Comment (Anupamaraie) [reply ciphers/vernam_cipher.py:3]: What is the change?

File: ciphers/vernam_cipher.py (hunk 2 above)
Comment (Anupamaraie) [reply ciphers/vernam_cipher.py]: what do i do?

**Comment by cclauss** [comment]
ruff and pre-commit both seem happy.

File: ciphers/vernam_cipher.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+# Example usage
+plaintext = "HELLO"
+key = "KEY"
+encrypted_text = vernam_encrypt(plaintext, key)
+decrypted_text = vernam_decrypt(encrypted_text, key)
+print("\n\n")
+print("Plaintext:", plaintext)
+print("Encrypted:", encrypted_text)
+print("Decrypted:", decrypted_text)
```
Comment (cclauss) [inline ciphers/vernam_cipher.py:44]:
```suggestion
if __name__ == "__main__":
    from doctest import testmod

    testmod()

    # Example usage
    plaintext = "HELLO"
    key = "KEY"
    encrypted_text = vernam_encrypt(plaintext, key)
    decrypted_text = vernam_decrypt(encrypted_text, key)
    print("\n\n")
    print("Plaintext:", plaintext)
    print("Encrypted:", encrypted_text)
    print("Decrypted:", decrypted_text)
```

Comment (cclauss) [reply ciphers/vernam_cipher.py:44]: This indentation is important so that the `input()` command is not run by pytest.

**Review by cclauss — APPROVED** [review]

File: ciphers/vernam_cipher.py (hunk 3 above)
Comment (Anupamaraie) [reply ciphers/vernam_cipher.py]: oh okay. I'll change it. Thank you

**Comment by cclauss** [comment]
It looks good now.

File: ciphers/vernam_cipher.py (hunk 1 above)
Comment (cclauss) [reply ciphers/vernam_cipher.py]: We only use this in `__main__`

**Comment by Anupamaraie** [comment]
Yes. Thank you so much for the help.

**Pull request merged**